src/util/chunked_content_decoder.js

```diff
@@ -82,6 +134,22 @@ class ChunkedContentDecoder extends stream.Transform {
                 } else {
                     this.state = STATE_READ_CHUNK_HEADER;
                 }
+            } else if (this.state === STATE_READ_TRAILER_DATA) {
+                const start_trailer = index;
+                for (; index < buf.length; ++index) {
+                    if (buf[index] === CR_CODE) {
+                        this.state = STATE_WAIT_NL_TRAILER;
+                        break;
+                    }
+                    this.chunk_trailer_str = buf.toString('utf8', start_trailer, index);
```

Comment:
since it might overlap a buffer, we need to append the string
```suggestion
                    this.chunk_trailer_str += buf.toString('utf8', start_trailer, index);
```

Reply:
also this needs to be outside the loop...